metadata-db-app/src/migrations/055_setup_services_table.clj

```diff
@@ -0,0 +1,58 @@
+(ns migrations.055-setup-services-table
+  (:require
+   [config.mdb-migrate-helper :as h]))
+
+(def ^:private services-column-sql
+  "id NUMBER,
+  concept_id VARCHAR(255) NOT NULL,
+  native_id VARCHAR(1030) NOT NULL,
+  metadata BLOB NOT NULL,
+  format VARCHAR(255) NOT NULL,
+  revision_id INTEGER DEFAULT 1 NOT NULL,
+  revision_date TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL,
+  created_at TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL,
+  deleted INTEGER DEFAULT 0 NOT NULL,
+  user_id VARCHAR(30),
+  service_name VARCHAR(20),
+  transaction_id INTEGER DEFAULT 0 NOT NULL")
+
+(def ^:private services-constraint-sql
+  (str "CONSTRAINT services_pk PRIMARY KEY (id), "
+       ;; Unique constraint on native id and revision id
+       "CONSTRAINT services_con_rev UNIQUE (native_id, revision_id)
+       USING INDEX (create unique index services_ucr_i ON cmr_services (native_id, revision_id)), "
+
+       ;; Unique constraint on concept id and revision id
+       "CONSTRAINT services_cid_rev UNIQUE (concept_id, revision_id)
+       USING INDEX (create unique index services_cri ON cmr_services (concept_id, revision_id))"))
+
+(defn- create-services-table
+  []
+  (h/sql
+   (format "CREATE TABLE METADATA_DB.cmr_services (%s, %s)"
+           services-column-sql services-constraint-sql)))
+
+
+(defn- create-services-indices
+  []
+  (h/sql "CREATE INDEX services_crdi ON METADATA_DB.cmr_services (concept_id, revision_id, deleted)")
```

Comment:
I've been thinking it might be a best practice to start documenting why we are creating specific indexes to make it easier for reviewers as well as to clarify for the author.

For example an index on just service_name would only fully satisfy the query "select service_name from cmr_services".

If we were actually issuing a query like "select concept_id from cmr_services where service_name = 'foo'" we'd want to include the concept_id as part of the index so that the index would fully cover the query.

I'm assuming the services table is small so it doesn't really matter, but for things like granules we should pay close attention.